Documentation/ceph-common-issues.md

```diff
@@ -684,3 +685,47 @@ Ceph CSI in its 1.2 version (with Rook 1.1) does not support the Erasure coded p
 
 So, if you are looking at using such storage class you should enable the Flex driver by setting `ROOK_ENABLE_FLEX_DRIVER: true` in your `operator.yaml`.
 Also, if you are in the need of specific features and wonder if CSI is capable of handling them, you should read [the ceph-csi support matrix](https://github.com/ceph/ceph-csi#support-matrix).
+
+## A worker node using RBD devices hangs up
+
+### Symptoms
+
+* There is no progress on I/O from/to one of RBD devices (`/dev/rbd*` or `/dev/nbd*`).
+* After that, the whole worker node hangs up.
+
+### Investigation
+
+In this case, you can see the following messages in `dmesg`.
+
+```console
+# dmesg
+...
+[51717.039319] INFO: task kworker/2:1:5938 blocked for more than 120 seconds.
+[51717.039361]       Not tainted 4.15.0-72-generic #81-Ubuntu
+[51717.039388] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
+...
+```
+
+It's so-called `hung_task` problem and means that there is a deadlock in the kernel. Here the following two logics cause the deadlock.
+
+* I/O to an RBD device: It allocates some memory in the kernel.
+* The kernel's memory reclaiming logic: It's kicked typically under low memory environment and incurs I/O to that device.
+
+The probability of this problem becomes high if the following conditions are satisfied.
+
+* XFS is created on the problematic RBD device.
+* This device and the corresponding OSDs are co-located.
+* There is high I/O pressure to this device.
+* There is lack of free memory in the worker node.
+* The Rook/Ceph cluster is under upgrading.
+
+For more detail, please refer to [the corresponding issue comment](https://github.com/rook/rook/issues/3132#issuecomment-580508760).
+
+### Solution
+
+This problem will be solve by the following two fixes.
+
+* Linux kernel: A minor feature that is introduced by [this commit](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=8d19f1c8e1937baf74e1962aae9f90fa3aeab463). It will be included in Linux v5.6.
+* Ceph: A fix that uses the above-mentioned kernel's feature. The Ceph community will probably discuss this fix after releasing Linux v5.6.
+
+In most cases, you can bypass this problem by using ext4 (or any other filesystems) rather than XFS. Although this problem might happen with other filesystems, it's possibility is very low.
```

Comment:
Can we specify the related csi option of the storage class for this?

Reply:
Should be `csi.storage.k8s.io/fstype`, but rook and ceph-csi default to ext4.

Reply:
Thanks,  I mentioned csi.storage.k8s.io/fstype in the new PR.